tests/integration/test_k8s.py

```diff
@@ -94,6 +94,36 @@ async def test_nodes_labelled(kubernetes_cluster):
     juju_nodes = [n for n in nodes if "juju-charm" in n["metadata"]["labels"]]
     assert len(k8s.units + worker.units) == len(juju_nodes), "Not all nodes labeled as juju-charms"
 
+async def test_coredns_integration(kubernetes_cluster):
```

Comment:
this test depends on the integration of coredns. 

```suggestion
async def test_coredns_integration(kubernetes_cluster, integrate_coredns):
```